### ISSUE TITLE
Missing tvOS Analytics dependency in AnalyticsWithoutAdIdSupport subspec

### ISSUE THREAD

**Issue by alexzchut** [firebase/firebase-ios-sdk#9779]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a GitHub issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.3.1
  * Firebase SDK version: 8.15.0
  * Installation method: `CocoaPods | Carthage | Zip file | Swift Package Manager` CocoaPods
  * Firebase Component: Analytics (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Target platform(s): `All | iOS | Catalyst | macOS | tvOS | watchOS` - tvOS

### [REQUIRED] Step 2: Describe the problem

Missing Analytics dependency when using AnalyticsWithoutAdIdSupport subspec for tvOS
 
#### Steps to reproduce:
When installing Firebase Analytics for tvOS, the dependent repository of FirebaseAnalytics is limited to iOS only and not installed when targeting for tvOS. 

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:

s.subspec 'AnalyticsWithoutAdIdSupport' do |ss|
    ss.ios.deployment_target = '10.0'
    ss.osx.deployment_target = '10.12'
    ss.tvos.deployment_target = '12.0'
    ss.ios.dependency 'FirebaseAnalytics/WithoutAdIdSupport', '~> 8.15.0'
    ss.dependency 'Firebase/CoreOnly'
  end

```
// TODO(you): code here to reproduce the problem
```

**Comment by paulb777** [comment]
@alexzchut Thanks for the issue and the PR! At first glance, this looks like an oversight and a good fix.

In the interim, a workaround should be to directly add `FirebaseAnalytics/WithoutAdIdSupport` to the Podfile.

**Comment by alexzchut** [comment]
@paulb777 yes, thats what I am actually doing for now, until this will be fixed in podspec

**Comment by tsunghung** [comment]
@alexzchut Thanks for the issue report and the fix!
